configure.ac

```diff
@@ -1636,21 +1620,12 @@ AS_IF([test "x$with_ldap" != "xno"],[
     AC_CHECK_HEADERS(mozldap/ldap.h)
     SQUID_CHECK_LDAP_API
   ])
-
-  AS_IF([test "x$LIBLDAP_LIBS" != "x"],[
-    CPPFLAGS="$LIBLDAP_CFLAGS $CPPFLAGS"
```

Comment:
This is intentional. CPPFLAGS are (supposed to be) only passed to the pre-compiler, whereas CXXFLAGS are passed to the C++ compiler (which may relay as-needed to the pre-compiler).

Reply:
> CPPFLAGS are (supposed to be) only passed to the pre-compiler, whereas CXXFLAGS are passed to the C++ compiler

The above statement is incorrect or misleading: Both CPPFLAGS and CXXFLAGS are supposed to be passed to the C++ compiler (because a C++ compiler includes preprocessing functionality -- it is both a preprocessor and a compiler of preprocessed sources). There is more on that in another change request, but the bottom line here is that we should not discard CPPFLAGS and should not copy them to CXXFLAGS. CPPFLAGS are a stand-alone entity.

Same for CPPGLAGS and CFLAGS.

Reply:
fixed in [d631d08](https://github.com/squid-cache/squid/pull/1333/commits/d631d084447b78440d4a71942eb34c1e739ddf5d)